### PR TITLE
fix: profile update fields (PLT-4581)

### PR DESCRIPTION
**Pull request by bogdan-manole**

### CODE REVIEW

File: src/Plutus/Certification/API/Routes.hs [hunk 1]
```diff
@@ -446,8 +447,34 @@ instance ToSchema DAppBody where
...
 
-instance ToSchema ProfileBody
+-- | A phantom type for website for swagger schema
+data Website
 
+instance ToSchema Website where
+  declareNamedSchema _ = do
+    return $ NamedSchema (Just "BaseUrl") $ mempty
+      & type_ ?~ SwaggerString
+      & SL.pattern ?~ ghAccessTokenPattern
```
Comment (RSoulatIOHK) [inline src/Plutus/Certification/API/Routes.hs:457]: Should be 
`$ SL.pattern ?~ baseUrlPattern`

File: src/Plutus/Certification/API/Routes.hs [hunk 2]
```diff
@@ -446,8 +447,34 @@ instance ToSchema DAppBody where
...
+    textSchema <- declareSchemaRef (Proxy :: Proxy Text)
+    dappSchema <- declareSchemaRef (Proxy :: Proxy DAppBody)
+    urlSchema <- declareSchemaRef (Proxy :: Proxy Website)
+    return $ NamedSchema (Just "Profile") $ mempty
+      & type_ ?~ SwaggerObject
+      & properties .~
+          [ ("dapp", dappSchema)
+          , ("website", urlSchema)
+          , ("vendor", textSchema)
+          , ("twitter", textSchema)
```
Comment (RSoulatIOHK) [inline src/Plutus/Certification/API/Routes.hs:473]: Couldn't you also use urlSchema for twitter and linkedin?

**Review by RSoulatIOHK — CHANGES_REQUESTED** [review]

File: src/Plutus/Certification/API/Routes.hs (hunk 1 above)
Comment (bogdan-manole) [reply src/Plutus/Certification/API/Routes.hs:457]: exactly 😔

Comment (bogdan-manole) [reply src/Plutus/Certification/API/Routes.hs]: done

File: src/Plutus/Certification/API/Routes.hs (hunk 2 above)
Comment (RSoulatIOHK) [reply src/Plutus/Certification/API/Routes.hs:531]: Good point. Yeah maybe this schéma for Twitter handle would be perfect

Comment (bogdan-manole) [reply src/Plutus/Certification/API/Routes.hs]: done

File: src/Plutus/Certification/API/Routes.hs [hunk 3]
```diff
@@ -265,11 +262,34 @@ instance ToJSON DAppBody where
...
+  )
+
+instance ToJSON Twitter where
+  toJSON (Twitter v) = String v
+
+instance ToSchema Twitter where
+  declareNamedSchema _ = do
+    return $ NamedSchema (Just "Twitter") $ mempty
+      & type_ ?~ SwaggerString
+      & SL.pattern ?~ "^[A-Za-z0-9_]{1,15}$"
```
Comment (RSoulatIOHK) [inline src/Plutus/Certification/API/Routes.hs:286]: Shouldn't we check if they added the "@" at the beginning? Something like: `^@?\w{1,15}$` ?

File: src/Plutus/Certification/API/Routes.hs [hunk 4]
```diff
@@ -265,11 +262,34 @@ instance ToJSON DAppBody where
...
+
 data ProfileBody = ProfileBody
    { dapp :: !(Maybe DAppBody)
-   , website :: !(Maybe Text)
+   , website :: !(Maybe BaseUrl)
    , vendor :: !(Maybe Text)
-   , twitter :: !(Maybe Text)
+   , twitter :: !(Maybe Twitter)
    , linkedin :: !(Maybe Text)
```
Comment (RSoulatIOHK) [inline src/Plutus/Certification/API/Routes.hs:293]: Can we also sanitize linkedin to ensure that if it is filled, it is indeed to a linkedin page?

If I give it a try (just for fun, not a recommendation in any way):
````haskell
linkedInPattern :: Pattern
-- | Straight outta ChatGPT
linkedInPattern = "^(?:https?:\/\/)?(?:www\.)?(?:linkedin\.com\/(?:in|pub|company)\/|lnkd\.in\/)(?:(?!.*?\/)[a-zA-Z0-9_-]+)(?:\/.*)?$" 


instance ToSchema LinkedIn where
		declareNamedSchema _ = do 
			return $ NamedSchema (Just "LinkedIn") $ mempty
			& type_ $~ SwaggerString
			& SL.pattern ?~ linkedInPattern
````

File: src/Plutus/Certification/API/Routes.hs (hunk 3 above)
Comment (bogdan-manole) [reply src/Plutus/Certification/API/Routes.hs:286]: For sure in the front, we should make such a constraint (@account_name) just to be sure the user understands, but for BE seems a bit redundant (the FE should remove the @ before calling the API).
I really don't have a strong opinion about this: your call

Comment (RSoulatIOHK) [reply src/Plutus/Certification/API/Routes.hs:286]: Good point to have the filtering done by front-end.

File: src/Plutus/Certification/API/Routes.hs (hunk 4 above)
Comment (bogdan-manole) [reply src/Plutus/Certification/API/Routes.hs]: done. please take a look at the latest commit 91b67b9d741c47511737ec7a720a53f11e384f9e

**Pull request merged**